src/main/java/io/swagger/inflector/controllers/SwaggerOperationController.java

```diff
@@ -409,6 +429,42 @@ else if(parameter.getRequired()) {
         }
         return Response.ok().build();
     }
+
+    public void validate(Object o, Property property, SchemaValidator.Direction direction) throws ApiException {
+        if(config.isValidatePayloads()) {
+            boolean isValid = SchemaValidator.validate(o, Json.pretty(property), direction);
+            if(!isValid) {
+                if(SchemaValidator.Direction.INPUT.equals(direction)) {
```

Comment:
Almost a year late :-) .. but maybe here is the best forum for my comment/ question. Currently, I do not see any way to push to the client the reason of a validation failure. Currently the client always gets back "Input does not match the expected structure", while it is reasonable to assume to get a more useful message.

Many thanks!